impeller/renderer/backend/vulkan/context_vk.cc

```diff
@@ -668,6 +671,7 @@ void ContextVK::InitializeCommonlyUsedShadersIfNeeded() const {
 }
 
 void ContextVK::DisposeThreadLocalCachedResources() {
+  Lock lock(desc_pool_mutex_);
```

Comment:
Do we need a lock in ~ContextVK, or do we have a guarentee that no one will be calling these from a separate thread while the contextvk is being deleted?

Reply:
hmm. I'm not sure.

Reply:
I think no, because the the destructor can only run once there are no more references to the context. So nothing else could be calling anything that references the desc pool? (unless done through a raw or weak ptr during destruction)

Reply:
If a `ContextVK` instance is being deleted, then the current thread is releasing the last reference to that instance and no other thread could be holding any references.

So there is no need to acquire any lock when the `cached_descriptor_pool_` member is destructed.

Reply:
Thank you Jason